skimage/feature/sift.py

```diff
@@ -0,0 +1,719 @@
+        sigma_in : float, optional
+            The assumed blur level of the input image.
+        c_dog : float, optional
+            Threshold to discard low contrast extrema in the DoG. It's final
+            value is dependent on n_scales by the relation:
+            final_c_dog = (2^(1/n_scales)-1) / (2^(1/3)-1) * c_dog
+        c_edge : float, optional
+            Threshold to discard extrema that lie in edges. If H is the
+            Hessian of an extremum, its "edgeness" is described by
+            tr(H)²/det(H). If the edgeness is higher than
+            (c_edge + 1)²/c_edge, the extremum is discarded.
+        n_bins : int, optional
+            Number of bins in the histogram that describes the gradient
+            orientations around keypoint.
+        lambda_ori : float, optional
+            The window used to find the reference orientation of a keypoint
+            has a width of 6 * lambda_ori * sigma and is weighted by a
+            standard deviation of 2 * lambda_ori * sigma.
+        c_max : float, optional
+            The threshold at which a secondary peak in the orientation
+            histogram is accepted as orientation
+        lambda_descr : float, optional
+            The window used to define the descriptor of a keypoint has a width
+            of 2 * lambda_descr * sigma * (n_hist+1)/n_hist and is weighted by
+            a standard deviation of lambda_descr * sigma.
+        n_hist : int, optional
+            The window used to define the descriptor of a keypoint consists of
+            n_hist * n_hist histograms.
+        n_ori : int, optional
+            The number of bins in the histograms of the descriptor patch.
+
+        Attributes
+        ----------
+        delta_min : float
+            The sampling distance of the first octave. It's final value is
+            1/upsampling.
+        float_dtype : type
+            The datatype of the image.
+        scalespace_sigmas : (n_octaves, n_scales + 3) array
+            The sigma value of all scales in all octaves.
+        keypoints : (N, 2) array
+            Keypoint coordinates as ``(row, col)``.
+        positions : (N, 2) array
+            Subpixel-precision keypoint coordinates as ``(row, col)``.
+        sigmas : (N, ) array
+            The corresponding sigma (blur) value of a keypoint.
+        sigmas : (N, ) array
+            The corresponding scale of a keypoint.
+        orientations : (N, ) array
+            The orientations of the gradient around every keypoint.
+        octaves : (N, ) array
+            The corresponding octave of a keypoint.
+        descriptors : (N, n_hist*n_hist*n_ori) array
+            The descriptors of a keypoint.
+
+        Notes
+        -----
+        The SIFT algorithm was developed by David Lowe [1]_, [2]_. The
+        implementation here closely follows the detailed description in [3]_,
+        including use of the same default parameters.
```

Comment:
Could you add a mention that it was patented, but that the patent expired?

https://patents.google.com/patent/US6711293B1/en